File: solutions/elixir/freelancer-rates/2/lib/freelancer_rates.ex
defmodule FreelancerRates do
  @daily_rate 8.0
  @monthly_billable_days 22
    
  def daily_rate(hourly_rate), do: hourly_rate * @daily_rate

  def apply_discount(before_discount, discount), do: before_discount * (1 - discount / 100)

  def monthly_rate(hourly_rate, discount) do
    daily_rate(hourly_rate) * @monthly_billable_days
    |> apply_discount(discount)
    |> ceil
  end

  def days_in_budget(budget, hourly_rate, discount) do
    budget / apply_discount(daily_rate(hourly_rate), discount)
    |> Float.floor(1)
  end

end

File: solutions/cpp/freelancer-rates/1/freelancer_rates.cpp
#include <cmath>

double daily_rate(double hourly_rate) {
    return hourly_rate * 8;
}

double apply_discount(double before_discount, double discount) {
    return before_discount * (1 - discount / 100.0);
}

int monthly_rate(double hourly_rate, double discount) {
    double monthly_without_discount = daily_rate(hourly_rate) * 22;
    double monthly = apply_discount(monthly_without_discount, discount);
    return ceil(monthly);
}

int days_in_budget(int budget, double hourly_rate, double discount) {
    return budget / apply_discount(daily_rate(hourly_rate), discount);
}